src/syncProvider.ts

```diff
@@ -146,6 +153,22 @@ export class FTPSyncProvider implements ISyncProvider {
         this.logger.verbose(`  file ${typePast}`);
     }
 
+    async syncMode(file: Record) {
+        if (!this.syncPosixModes) {
+            return;
+        }
+        this.logger.verbose("Syncing posix mode for file " + file.name);
+        // https://www.martin-brennan.com/nodejs-file-permissions-fstat/
+        let stats = await stat(this.localPath + file.name);
+        let mode: string = "0" + (stats.mode & parseInt('777', 8)).toString(8);
+        // https://github.com/patrickjuchli/basic-ftp/issues/9
+        let command = "SITE CHMOD " + mode + " " + file.name
+        if (this.dryRun === false) {
+            await this.client.ftp.request(command);
+        }
+        this.logger.verbose("Setting file mode with command " + command);
```

Comment:
This method is correct, but not sure if this will work, because you are only calling the `syncMode` function when you have differences in folder/files.

What if you move your change to another place? or change the perspective.

For example:
* I would improve the interfaces [IFile](https://github.com/SamKirkland/ftp-deploy/blob/f79bd104c89cbbf01c51e48cdd08efa522074e39/src/types.ts#L82-L87) and [IFolder](https://github.com/SamKirkland/ftp-deploy/blob/f79bd104c89cbbf01c51e48cdd08efa522074e39/src/types.ts#L89-L93) with a new attribute (i.e. `chmod`)
* When we are loading the local files [getLocalFiles](https://github.com/SamKirkland/ftp-deploy/blob/f79bd104c89cbbf01c51e48cdd08efa522074e39/src/localFiles.ts#L11-L30), you need to process the chmod attribute too.
* When the folders/files [are the same](https://github.com/SamKirkland/ftp-deploy/blob/f79bd104c89cbbf01c51e48cdd08efa522074e39/src/HashDiff.ts#L75), you need to add the chmod comparison.
* Also maybe, I would add a new list in [DiffResult](https://github.com/SamKirkland/ftp-deploy/blob/f79bd104c89cbbf01c51e48cdd08efa522074e39/src/types.ts#L110-L112) with only `chmod` changes.

I guess this is a better approach for your problem, and this is according with `ftp-deploy` project with the diff files comparison approach.

@SamKirkland thoughts?

Reply:
@JaviRpo That could be a fine idea. The correct name for such an attribute wouldn't be `chmod` but `mode` ;) `chmod` stands for "change mode".